lib/address.h

```diff
@@ -0,0 +1,77 @@
+/**
+ * @file
+ * Representation of an email address
+ *
+ * @authors
+ * Copyright (C) 2017 Richard Russon <rich@flatcap.org>
+ *
+ * @copyright
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, either version 2 of the License, or (at your option) any later
+ * version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _LIB_ADDRESS_H
+#define _LIB_ADDRESS_H
+
+#include <stdbool.h>
+#include <stdio.h>
+
+extern const char RFC822Specials[];
+extern int RFC822Error;
+extern const char *const RFC822Errors[];
+
+#define rfc822_error(x) RFC822Errors[x]
+
+/**
+ * struct Address - An email address
+ */
+struct Address
+{
+  char *personal; /**< real name of address */
+  char *mailbox;  /**< mailbox and host address */
+  int group;      /**< group mailbox? */
+  struct Address *next;
```

Comment:
Definitely electable to a switch to TAILQ (this comment is more a note than an actual request for change)

Reply:
> Definitely electable to a switch to TAILQ

Yep.

With independent libraries, it should be easy for someone to make and test these changes.
(someone who needn't be familiar with neomutt's code).